tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -382,98 +382,152 @@ def search_versions(filter_string):
+            self._search_registered_models("name!=something")
         assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
         # cannot search by run_id
         with self.assertRaises(MlflowException) as exception_context:
-            search_registered_model("run_id='%s'" % "somerunID")
+            self._search_registered_models("run_id='%s'" % "somerunID")
         assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
         # cannot search by source_path
         with self.assertRaises(MlflowException) as exception_context:
-            search_registered_model("source_path = 'A/D'")
+            self._search_registered_models("source_path = 'A/D'")
         assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
         # cannot search by other params
         with self.assertRaises(MlflowException) as exception_context:
-            search_registered_model("evilhax = true")
+            self._search_registered_models("evilhax = true")
         assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
-        # delete rm6. search should not return RM 6
-        self.store.delete_registered_model(name=name6)
+        # delete rm6. search should not return RM 5
+        self.store.delete_registered_model(name=names[-1])
 
         # equality search using name should return no names
-        self.assertEqual(set(search_registered_model("name='%s'" % name6)), set([]))
+        self.assertEqual(set(self._search_registered_models(
+            f"name='{names[-1]}'")), set([]))
 
         # case-sensitive prefix search using LIKE should return all the RMs
-        self.assertEqual(set(search_registered_model(f"name LIKE '{prefix}%'")),
-                         set([name1, name2, name3, name4, name5]))
+        self.assertEqual(set(self._search_registered_models(
+            f"name LIKE '{prefix}%'")), set(names[0:5]))
 
         # case-insensitive prefix search using ILIKE should return both rm5 and rm6
-        self.assertEqual(set(search_registered_model(f"name ILIKE '{prefix + 'RM4A'}%'")),
-                         set([name5]))
+        self.assertEqual(set(self._search_registered_models(
+            f"name ILIKE '{prefix + 'RM4A'}%'")), {names[4]})
+
+    def test_search_registered_model_pagination(self):
+        rms = [self._rm_maker(f"RM{i:03}").name for i in range(50)]
+
+        # test flow with fixed max_results
+        returned_rms = []
+        result = self._search_registered_models(
+            "name LIKE 'RM%'", page_token=None, max_results=5)
+        returned_rms.extend(result)
+        while result.token:
+            result = self._search_registered_models(
+                "name LIKE 'RM%'", page_token=result.token, max_results=5)
+            returned_rms.extend(result)
+        self.assertEqual(set(rms), set(returned_rms))
```

Comment:
Why did you need to convert this list to a set? Would be beneficial to compare with expected ordering as well.

Reply:
Yup - fixed